client/state/reader/related-posts/reducer.js

```diff
@@ -15,41 +10,39 @@ import {
 } from 'calypso/state/reader/action-types';
 import { key } from './utils';
 
-export const items = withoutPersistence( ( state = {}, action ) => {
+function setStateForKey( state, action, val ) {
+	const { siteId, postId, scope } = action.payload;
+	return {
+		...state,
+		[ key( siteId, postId, scope ) ]: val,
+	};
+}
+
+export const items = ( state = {}, action ) => {
 	switch ( action.type ) {
-		case READER_RELATED_POSTS_RECEIVE: {
-			state = assign( {}, state, {
-				[ key( action.payload.siteId, action.payload.postId, action.payload.scope ) ]: map(
-					action.payload.posts,
-					'global_ID'
-				),
-			} );
-			return state;
-		}
+		case READER_RELATED_POSTS_RECEIVE:
+			return setStateForKey(
+				state,
+				action,
+				action.payload.posts.map( ( p ) => p.global_ID )
```

Comment:
I've verified that `action.payload.posts` will always be an array, so we're safe here.